media/Jsontracer/index.js

```diff
@@ -0,0 +1,195 @@
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// This file referenced https://github.com/catapult-project/catapult/tree/master/tracing
+
+// TODO import openFileSelector from "./processData.js";
+// TODO import dynamicGraduation from "./dynamicGraduation.js";
+
+const graph = document.querySelector(".graph");
+
+const loadBtn = document.querySelector(".load-btn");
+loadBtn.addEventListener("click", () => {
+  initData();
+  // TODO openFileSelector();
+});
+
+let ratio = 100;
+const slider = document.querySelector("input");
+slider.addEventListener("input", event => {
+  ratio = event.target.value * 1;
+  graph.style.width = `${ratio}%`;
+  changeSlider(event.target.value, event.target.max, event.target.min);
+  // TODO dynamicGraduation();
+});
+
+const zoomInBtn = document.querySelector(".zoom-in-btn");
+zoomInBtn.addEventListener("click", () => {
+  if (ratio >= 5000) { return; }
+  
+  // change ratio
+  ratio += 50;
+  ratio = ratio > 5000 ? 5000 : ratio;
+  graph.style.width = `${ratio}%`;
+
+  // change slider
+  const slider = document.querySelector("input");
+  slider.value = ratio;
+  changeSlider(slider.value, slider.max, slider.min);
+
+  // change graduation and set delay
+  dynamicGraduation();
+  zoomInBtn.disabled = true;
+  setTimeout(() => (zoomInBtn.disabled = false), 300);
+});
+
+const zoomOutBtn = document.querySelector(".zoom-out-btn");
+zoomOutBtn.addEventListener("click", () => {
+  if (ratio <= 100) { return; }
+
+  // change ratio
+  ratio -= 50;
+  ratio = ratio < 100 ? 100 : ratio;
+  graph.style.width = `${ratio}%`;
+
+  // change slider
+  const slider = document.querySelector("input");
+  slider.value = ratio;
+  changeSlider(slider.value, slider.max, slider.min);
+
+  // change graduation and set delay
+  dynamicGraduation();
+  zoomOutBtn.disabled = true;
+  setTimeout(() => (zoomOutBtn.disabled = false), 300);
+});
+
+const captureBtn = document.querySelector(".capture-btn");
+captureBtn.addEventListener("click", () => {
+  // TODO capture inside vscode extension webview
+});
+
+function changeSlider(inputValue, inputMax, inputMin) {
+  if (inputMax === inputValue) {
+    if (inputMax === "200") {
+      slider.max = "400";
+      slider.value = "200";
+      slider.min = "100";
+    } else if (inputMax === "400") {
+      slider.max = "800";
+      slider.value = "400";
+      slider.min = "200";
+    } else if (inputMax === "800") {
+      slider.max = "1600";
+      slider.value = "800";
+      slider.min = "400";
+    } else if (inputMax === "1600") {
+      slider.max = "3200";
+      slider.value = "1600";
+      slider.min = "800";
+    } else if (inputMax === "3200") {
+      slider.max = "5000";
+      slider.value = "3200";
+      slider.min = "1600";
+    }
```

Comment:
Can we apply following codes?

```suggestion
      const SliderMaxLimit = 5000; // Define at the top of the code

      slider.max = String(Math.max(inputMax*2, SliderMaxLimit));
      slider.value = String(inputMax);
      slider.min = String(inputMax*0.5);
```

Reply:
Thank you for the feedback! I'll change it like this.

```
if (inputMax === inputValue) {
    const sliderMaxLimit = 6400; // Define at the top of the code
    slider.max = String(Math.min(inputMax*2, sliderMaxLimit));
    slider.value = String(inputMax);
    slider.min = String(inputMax*0.5);
  } else if (inputMin === inputValue) {
    const sliderMinLimit = 100; // Define at the top of the code
    slider.max = String(inputMin*2);
    slider.value = String(inputMin);
    slider.min = String(Math.max(inputMin*0.5, sliderMinLimit));
  } else {
    return;
  }  
```

`sliderMaxLimit` , `sliderMinLimit` I'll place these at the top. :)